tests/src/Kernel/Validators/ValidatorGermplasmNameExistsTest.php

```diff
@@ -0,0 +1,449 @@
+      ->execute();
+    $this->assertIsNumeric($this->organism_id, 'We were not able to create the organism ' . $genus . ' ' . $species . ' in Chado for testing.');
+
+    // Insert a single germplasm into chado.stock.
+    $values = [
+      'organism_id' => $this->organism_id,
+      'name' => $this->inserted_germplasm_name,
+      'uniquename' => 'TEST:1',
+      'type_id' => 9,
+    ];
+    $stock_id = $this->chado_connection->insert('1:stock')
+      ->fields($values)->execute();
+    $this->assertIsNumeric($stock_id, 'We were not able to create the stock ' . $this->inserted_germplasm_name . ' in Chado for testing.');
+
+    // Insert our set of duplicate germplasm names into chado.stock.
+    foreach ($this->duplicate_insert_values as $key => $duplicate) {
+      $stock_id = $this->chado_connection->insert('1:stock')
+        ->fields($duplicate)->execute();
+      $this->assertIsNumeric($stock_id, 'We were not able to create the stock with uniquename ' . $duplicate['uniquename'] . ' in Chado for testing.');
+      // Update our array with the stock_id.
+      $this->duplicate_insert_values[$key]['stock_id'] = $stock_id;
+    }
+  }
+
+  /**
+   * Data Provider: Provide different row values with germplasm names.
+   *
+   * @return array
+   *   Each test scenario is an array with the following values:
+   *   - An array where each value is the key of the column in $row_values that
+   *     the validator instance should act on.
+   *   - An array of values from a single row/line in the file where each key
+   *     maps to a column index and each value is the content of that column.
+   *   - An array of the expected contents of the returned validation result:
+   *     - 'expected_valid': The expected validation status (TRUE if pass, FALSE
+   *       fail)
+   *     - 'expected_case': The expected case message.
+   *     - 'expected_failedItems': The expected contents of the 'failedItems'
+   *       array with the following keys. This should be an empty array if
+   *       validation is expected to pass.
+   *       - 'missing_cells' (OPTIONAL): Present if a germplasm name is missing
+   *         from the database.
+   *         - 1+ arrays keyed by the column number (first column = 1) in the
+   *          input row that contains a missing germplasm name, further keyed
+   *          by:
+   *          - 'germplasm_name': The name of the missing germplasm.
+   *       - 'duplicate_cells' (OPTIONAL): Present if a germplasm name is
+   *         duplicated in the database.
+   *         - 1+ arrays keyed by the column number (first column = 1) in the
+   *          input row that contains a duplicate germplasm name, further keyed
+   *          by:
+   *          - 'germplasm_name': The name of the duplicate germplasm.
+   *          - 'duplicates': A list of indices in $duplicate_insert_values
+   *            which correspond to records that are expected to be duplicated.
+   */
+  public function provideRowToGermplasmNameExists() {
```

Comment:
It would be good to add a test scenario here where the germplasm name is empty. That said, we still need to decide what we expect in such a scenario :see_no_evil:

Reply:
Addressed in commit 42bb27707600cc5ff88d5d284e7e7996a5b9f0e2